### DISCUSSION TITLE
Transparent sqare

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#3737] [Help]
How to extend a rectangle to a square by a transparent background. It should work for any size with the same script.

**Comment by fmw42** [comment]
Are you asking about a rectangle image or a rectangle within an image?  What version of ImageMagick 6 or 7? What platform/OS?

**Reply by fmw42** [reply]
Input:

![barn](https://user-images.githubusercontent.com/13458218/119401007-c405ed00-bc8f-11eb-921f-5d61a002f09f.jpg)

`magick barn.jpg -background none -extent "%[fx:max(w,h)]x%[fx:max(w,h)]" barn_extent.png`

Result:

![barn_extent](https://user-images.githubusercontent.com/13458218/119401064-d6802680-bc8f-11eb-93b5-ca89c270d04e.png)

```
magick identify barn.jpg barn_extent.png
barn.jpg JPEG 400x299 400x299+0+0 8-bit sRGB 51909B 0.000u 0:00.000
barn_extent.png PNG 400x400 400x400+0+0 8-bit sRGB 254868B 0.000u 0:00.000
```

**Reply by tbmuc** [reply]
Looks good .how to center it in order have the same space on top and at the bottom? With gravity center before the output?

**Reply by fmw42** [reply]
Yes, before -extent

`magick barn.jpg -background none -gravity center -extent "%[fx:max(w,h)]x%[fx:max(w,h)]" barn_extent.png`
